src/crypto/equihash.cpp

```diff
@@ -39,6 +39,85 @@ int Equihash<N,K>::InitialiseState(eh_HashState& base_state)
                                                          personalization);
 }
 
+void ExpandArray(const unsigned char* in, size_t in_len,
+                 unsigned char* out, size_t out_len,
+                 size_t bit_len)
+{
+    assert(bit_len >= 8);
+    assert(8*sizeof(uint32_t) >= 7+bit_len);
+
+    size_t out_width { (bit_len+7)/8 };
+    assert(out_len == 8*out_width*in_len/bit_len);
```

Comment:
A case where the assertion would incorrectly fail is n = 144, k = 7. In that case bit_len = n/(k+1)+1 = 19, out_width = 3, and 2^k = 128, but 8*3*128/19 != (8*128/19)*3.